State.cpp

```diff
@@ -56,4 +56,14 @@ bool State::operator==(State const &other)
     return this->name           ==  other.name              &&
            this->transition_map ==  other.transition_map    &&
            this->type           ==  other.type  ;
-}
\ No newline at end of file
+}
+
+std::ostream& operator<<(std::ostream& strm, State const& state)
+{
+    strm << "State Name: " <<  state.name << "\n" << "State Type: " << state.type << "\n"
```

Comment:
what if we just mapped enum values as keys to type in string values in a hashmap to make it look less clunky

Reply:
Ya we can could an array/vector of strings or a map, but we should not declare it as a global variable because I would think that is bad practice?

Instead, we can overload the operator << for the `enum StateType` using one of the three answers in https://stackoverflow.com/questions/3342726/c-print-out-enum-value-as-text/3342891. Personally, I like the switch-case answer so that we don't have to allocate memory for a map even though its only temporary.

Reply:
also once we convert it, it will look like for example: "State Type: normalstatetype" which sounds weird so maybe we change it to just "normal" and "starting" and "end"

Reply:
honestly easiest solution is to make state type a string...